modules/mail.nix

```diff
@@ -0,0 +1,299 @@
+          description = "Just to identify the account";
+        };
+
+        userName = mkOption {
+          type = types.str;
+          description = "The foreground color.";
+        };
+
+        realname = mkOption {
+          type = types.str;
+          description = "Name displayed when sending mails.";
+        };
+
+        # make into a submodule ? attach signature ?
+        signatureFilename = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+          example = "";
+          description = "Path to signature";
+        };
+
+        showSignature = mkOption {
+          # TODO enum attach/append
+          type = types.enum [ "append" "attach" "no" ];
+          default = "no";
+          description = "wether to attach signature";
+        };
+
+        # identity ? at least of alot
+        gpgKey = mkOption {
+          type = types.nullOr types.string;
+          default = null;
+          description = "your gpg key";
+        };
+
+        address = mkOption {
+          type = types.str;
+          example = "luke@tatooine.com";
+          description = "Your mail address";
+        };
+
+        imapHost = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "imap.server.be";
+          description = "IMAP host to receive mails from";
+        };
+
+        sendHost = mkOption {
+          type = types.str;
+          example = "luke@tatooine.com";
+          description = "MSMTP host to use to send mails";
+        };
+
+        # see http://alot.readthedocs.io/en/latest/configuration/contacts_completion.html
+        contactCompletion = mkOption {
+          # TODO add it only if notmuch available ?
+          type = types.enum [ "notmuch address simple" "notmuch address" ];
+          default = "notmuch address";
+          description = "path to the hooks folder to use for a specific account";
+        };
+
+        # default actions
+        defaultActions = mkOption {
+          # TODO add it only if notmuch available ?
+          type = types.listOf (types.enum [ "cryptoSign" "cipher" "appendSignature" "attachSignature" ]);
+          default = false;
+          description = "Wether to sign messages";
+        };
+
+        # can have only one mta
+        mra = mkOption {
+          type =  types.enum [config.programs.mbsync config.programs.offlineimap];
+          default = config.programs.offlineimap;
+          description = "Mail Retrieval Agent to use";
+        };
+
+        mta = mkOption {
+          type =  types.enum [config.programs.msmtp];
+          default = config.programs.msmtp;
+          description = "Mail Transfer Agent to use";
+        };
+
+        MUAs = mkOption {
+          # TODO add astroid
+          type = types.listOf (types.enum [ config.programs.alot  ]);
+          default = [ config.programs.alot ];
+          description = "List of Mail User Agents to take into account";
+        };
```

Comment:
Instead of having MUAs/mra/mta defined here, it might be best to have mail-related programs take a list of mailAccount as a parameter ?